### PR TITLE
Fix uniqueness0 issue and MLP dropout issue

### PR DESCRIPTION
**Pull request by Nano1337**
## Uniqueness0 issue in synthetic dataset: 

It seems that when the synthetic data was retrieved through the __getitem__() method of SyntheticDataset(nn.Dataset) for uniqueness0 it was defaulting to a ones tensor that was written to handle missing keys in the data dictionary. However, even though '0' was a valid key, doing "if modality" when modality=0 in Python results in a False evaluation, which causes the uniqueness0 tensor to just be a ones tensor. This greatly affected downstream results for me, and I would recommend checking if this happens in the results reported into the actual paper by manually checking the contents of the first batch in the train dataloader. If this is the case, then I would highly recommend re-running all synthetic dataset experiments for this section. 

## MLP Dropout issue: 

the dropout wasn't being applied to the correct output tensor, i.e. the second dropout was being applied to the first tensor again, which resulted in no dropout for the actual output. This shouldn't affect results too much but I would re-run and see how it actually affects results where dropout is used for MLP. 

## Reproducibility Suggestion:
For other suggestions not found in this PR, I ensured result reproducibility by using pytorch-lightning's seed_everything function as follows: 
`
from pytorch_lightning import seed_everything
seed_everything(args.seed)
`
and backend settings for deterministic algos:
`
torch.backends.cudnn.deterministic = True
torch.backends.cudnn.benchmark = False
`

### CODE REVIEW

**Comment by kapikantzari** [comment]
Hi, sorry about the late reply. You are indeed correct about the modality argument - we originally created the synthetic data using an older version of the script (which uses True/False to indicate whether to use each modality). However, when we later updated the code to be the current version for better readability, we overlooked this line in loading the data. Sorry about the confusion - your suggested changes make sense and thank you for catching that!
